source/config.go

```diff
@@ -71,20 +75,21 @@ func Parse(cfg map[string]string) (Config, error) {
 	}
 
 	sourceConfig := Config{
-		Config:         common,
-		StreamName:     cfg[ConfigKeyStreamName],
-		Durable:        cfg[ConfigKeyDurable],
-		DeliveryPolicy: cfg[ConfigKeyDeliveryPolicy],
-		AckPolicy:      cfg[ConfigKeyAckPolicy],
+		Config:     common,
+		StreamName: cfg[ConfigKeyStreamName],
+		Durable:    cfg[ConfigKeyDurable],
 	}
 
-	if cfg[ConfigKeyBufferSize] != "" {
-		bufferSize, err := strconv.Atoi(cfg[ConfigKeyBufferSize])
-		if err != nil {
-			return Config{}, fmt.Errorf("\"%s\" must be an integer", ConfigKeyBufferSize)
-		}
+	if err := parseBufferSize(cfg[ConfigKeyBufferSize], &sourceConfig); err != nil {
+		return Config{}, fmt.Errorf("parse buffer size: %w", err)
+	}
 
-		sourceConfig.BufferSize = bufferSize
+	if err := parseDeliverPolicy(cfg[ConfigKeyDeliverPolicy], &sourceConfig); err != nil {
```

Comment:
Like proposition maybe move those functions to config methods

Reply:
Done